src/eventloop.c

```diff
@@ -203,9 +207,11 @@ guint input_add(int fd, PurpleInputCondition cond,
     input_event->parent = input_handle;
     input_handle->events = g_list_append(input_handle->events, input_event);
 
+    input_event->id = evLoopState.last_event_id++;
```

Comment:
Should probably check if we didn't overflow here – how knows for how many decades it can run now that it got as stable as it is ;)

Reply:
we can check if `evLoopState.last_event_id` is maxint if you like, but honestly...meeeh